src/Config/ObjectModel/RuntimeEntities.cs

```diff
@@ -56,7 +56,21 @@ public bool ContainsKey(string key)
         return Entities.ContainsKey(key);
     }
 
-    public Entity this[string key] => Entities[key];
+    public Entity this[string key]
+    {
+        get
+        {
+            Entity? entity;
+            if (Entities.TryGetValue(key, out entity) && entity is not null)
+            {
+                return entity;
+            }
+            else
+            {
+                throw new ApplicationException($"The entity '{key}' was not found in the dab-config json");
```

Comment:
why ApplicationException and not DataApiBuilderException which is used throughout the project?

Reply:
It makes sense, Since, it is a small change, making this change in patch PR

Reply:
@sajeetharan what do you think?

Reply:
also message should be something like:
"... was not found in the runtime config."

using "dab-config json" is too close to "dab-config.json" which is an explicit file name. End-users most likely aren't using `dab-config.json` as the file name (more likely dab-config.mssql.json, dab-config.cosmos.json etc.) so this may add confusion.